docker/init.sls

```diff
@@ -48,13 +48,35 @@ docker-config:
     - mode: 644
     - user: root
 
+{% if docker.daemon_config %}
+docker-daemon-dir:
+  file.directory:
+    - name: /etc/docker
+    - user: root
+    - group: root
+    - mode: 755
+
+docker-daemon-config:
+  file.serialize:
+    - name: /etc/docker/daemon.json
+    - user: root
+    - group: root
+    - mode: 644
+    - dataset:
+        {{ docker.daemon_config | yaml() | indent(8) }}
+    - formatter: json
+{% endif %}
+
 docker-service:
   service.running:
     - name: docker
     - enable: True
     - watch:
       - file: /etc/default/docker
       - pkg: docker-package
+      {% if docker.daemon_config %}
+      - file: /etc/docker/daemon.json
```

Comment:
I think this is wrong, as `watch` should point to *states* to watch, not 'files' themselves. So in this case, it should be

```yaml
- file: docker-daemon-config
```

As is, the formula won't fail but also won't act on the `file` state.

The same error is on the `watch`ed file above, on line 75.

Reply:
Both `file` and `state-name` are accepted by Salt afaik.  I tend to use state name. 

But a related observation is that we have hardcoding of filename in two places in this SLS file. The file could be hardcoded in `defaults.yaml` and `pillar.example` instead of SLS file.

Reply:
@noelmcloughlin, I check [the docs](https://docs.saltstack.com/en/latest/ref/states/requisites.html#watch) before commenting, because was not sure if that was possible, and all I see in the documentation is that the `state-name` should be used. That's why I requested the changes, to be sure we do it the 'documented-way' (besides the hard-coding, ofc).

Reply:
@javierbertoli I would like it changed too. state-name feels more correct. yes, lets remove hardcoding before merging. thanks !!

Reply:
I always use state-name but there is lot off `file:` in this formula, I decided to continue in the same way

Reply:
Would it be a lot of trouble for you to change them to state-names, as discussed above, so we remove hard-coding in this PR?

Reply:
Should I change to 
```
    - watch:
      - file: docker-config
      - pkg: docker-package
      {% if docker.daemon_config %}
      - file: docker-daemon-config
      {% endif %}
```
or 
```
    - watch:
      - docker-config
      - docker-package
      {% if docker.daemon_config %}
      - docker-daemon-config
      {% endif %}
```

Reply:
I usually go with the full format (`state_module: id`), as I find it easier to spot the dependency when reading the code, but both are valid, the latter since [2016.3.0](https://docs.saltstack.com/en/latest/ref/states/requisites.html#omitting-state-module-in-requisites). I think there are no guidelines about these, so do it as you please, as you're doing the hard work. :+1: